thunx-predicates-querydsl/src/main/java/com/contentgrid/thunx/predicates/querydsl/FieldByReflectionAccessStrategy.java

```diff
@@ -0,0 +1,45 @@
+package com.contentgrid.thunx.predicates.querydsl;
+
+import com.querydsl.core.util.ReflectionUtils;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import lombok.NonNull;
+import lombok.Value;
+
+public class FieldByReflectionAccessStrategy implements PropertyAccessStrategy {
+
+    @Override
+    public Optional<PropertyAccess> getProperty(Class<?> type, String pathElement) {
+
+        return Optional.ofNullable(ReflectionUtils.getFieldOrNull(type, pathElement))
+                .map(FieldPropertyAccess::new);
+    }
+
+    @Value
+    static class FieldPropertyAccess implements PropertyAccess {
```

Comment:
Shouldn't this be a private class, with nothing exposed?

The `PropertyAccess` interface should be enough to do everything that is necessary.